### ISSUE TITLE
[LOW] [Splits] IOU - IOU becomes paid while receipt is scanning

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#33972]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.21-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Send an IOU from Account A to Account B
2. Open Account B and send a receipt IOU
3. While the IOU is scanning click on pay elsewhere
4. Click on the preview component to open the IOU conversation and observe the IOU while receipt is being scanned

## Expected Result:
Receipt IOU should not be marked as paid

## Actual Result:
The receipt IOU is marked as paid before the receipt is done scanning and before account B marks it as paid

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] Windows: Chrome
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/e56b724f-66d9-413d-894a-58ea09fc7739

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a008da44d804c493</li>
        <li>Upwork Job ID: 1742972707793920000</li>
        <li>Last Price Increase: 2024-01-25</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @youssef-lr</details>

**Comment by Tony-MK** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

IOU becomes paid while the receipt is scanning.

### What is the root cause of that problem?

The amount to be paid in `iouReport` depends on the amount of the receipt. The settlement button should be displayed if the `reimbursableTotal !== 0`. 

However, the `ReportPreview` and `MoneyReportHeader` components will display the settlement button even if the value `reimburasbleTotal` will not be correct. 

This is because the receipt's amount is incorrect due to it not being processed yet.

https://github.com/Tony-MK/Expensify-App/blob/d5741afd6762560d542bdb1b82139eee1e787f12/src/components/ReportActionItem/ReportPreview.js#L253

### What changes do you think we should make in order to solve the problem?


Therefore, let's check if a receipt has its amount missing to improve the condition to determine the boolean value of `shouldShowSettlementButton`. 

This will ensure the button will displayed when the accurate `reimbursableTotal` is refelected. 

Thus, the `iouReport` can avoid the extra invalid `reportActions` will be created if the `iouReport` is paid. 

In case the user enters an amount in the receipt or it completes scanning, the settlement button will become visible and active.

We can modify the condition `shouldShowSettlementButton` for the `ReportPreview` and `MoneyReportHeader` to something similar to the one below.

```
const shouldShowSettlementButton =  (shouldShowPayButton || shouldShowApproveButton) && !_.some(
    transactionsWithReceipts, 
    (transaction) => TransactionUtils.isReceiptBeingScanned(transaction) && TransactionUtils.isAmountMissing(transaction)
);
```

### What alternative solutions did you explore? (Optional)

We could use the `isDisabled` prop for the settlement button on this [line](https://github.com/Tony-MK/Expensify-App/blob/d5741afd6762560d542bdb1b82139eee1e787f12/src/components/ReportActionItem/ReportPreview.js#L319) for the `ReportPreview` component.

For example:
```
<SettlementButton
   isDisabled={_.some(
        transactionsWithReceipts, 
        (transaction) => TransactionUtils.isReceiptBeingScanned(transaction) && TransactionUtils.isAmountMissing(transaction)
    )}
```

**Comment by garrettmknight** [comment]
@Ollyws can you please review the proposals so far?

**Comment by Tony-MK** [comment]
Thank you, @Ollyws, for reviewing my proposal, and I apologize for failing to grasp the expected result.

I have updated my proposal based on your comments at https://github.com/Expensify/App/issues/33972#issuecomment-1887569945.

However, there are some questions which I would appreciate if you answer.

Thank you.

**Comment by abzokhattab** [comment]
Thanks @Ollyws  added a second solution to avoid marking scanning receipts as paid.

**Comment by garrettmknight** [comment]
@Ollyws can you review the updates when you get a chance?

**Comment by Tony-MK** [comment]
@Ollyws, Thanks for the heads up. I didn't notice.

Then, I think using `TransactionUtils.isReceiptBeingScanned(transaction)` in the `canEditFieldOfMoneyRequest` and `canEditMoneyRequest` functions should be enough to render the report as editable.

**Comment by Tony-MK** [comment]
Thanks for the heads up, @Ollyws.

Then, I think using `TransactionUtils.isReceiptBeingScanned(transaction)` in the `canEditFieldOfMoneyRequest` and `canEditMoneyRequest` functions should be enough to render the report as editable.

What do you think? 🤔

**Comment by garrettmknight** [comment]
Checking with VIP split to see if this fits there: https://expensify.slack.com/archives/C05RECHFBEW/p1707162631162859

**Comment by garrettmknight** [comment]
Bumped VIP again

**Comment by garrettmknight** [comment]
Shouldn't be overdue

**Comment by garrettmknight** [comment]
Adding to vip-split - @youssef-lr can you give this one a look and confirm it needs to be internal?

**Comment by youssef-lr** [comment]
Going to take a look today @garrettmknight!

**Comment by garrettmknight** [comment]
@youssef-lr just assigning you to confirm internal v. external. If it's internal we can flag it for all wave7 core.

**Comment by garrettmknight** [comment]
@dylanexpensify just reassigning in case this needs help while I'm OOO this week - will be back 2/21 to pick it back up.

**Comment by youssef-lr** [comment]
@garrettmknight @dylanexpensify I'll be able to confirm today if this needs to be internal.

**Comment by youssef-lr** [comment]
I'm able to reproduce and I think this does need to be internal.

**Comment by dylanexpensify** [comment]
nice, thanks @youssef-lr ❤️

**Comment by dylanexpensify** [comment]
@youssef-lr are you taking this on? If so, let's remove hot pick label! 😄

**Comment by garrettmknight** [comment]
Just assigned @youssef-lr originally to confirm internal or not - leaving hot-pick on for now, but @youssef-lr if you want to pick it up, feel free to assign yourself!

**Comment by garrettmknight** [comment]
Waiting on hot picks!

**Comment by garrettmknight** [comment]
Still waiting

**Comment by garrettmknight** [comment]
Still waiting

**Comment by youssef-lr** [comment]
I'll work on this next week @garrettmknight

**Comment by garrettmknight** [comment]
My man! 🤝

**Comment by youssef-lr** [comment]
Will start investigation tomorrow, I'm not sure I can get to this today.

**Comment by youssef-lr** [comment]
@garrettmknight I think we can hold this on https://github.com/Expensify/App/issues/35240. The solution is to move the smartscanning expense to a new report. I'm wondering though if #35240 will support IOUs or just expenses? cc @mountiny

**Comment by garrettmknight** [comment]
I'm not sure - @jasperhuangg might be able to help us understand that too.

**Comment by mountiny** [comment]
I think it should work same for both. We can make it clearer in the issue if @jasperhuangg agrees

**Comment by youssef-lr** [comment]
Added the hold accordingly @garrettmknight @mountiny

**Comment by jasperhuangg** [comment]
Hey! Sorry for missing these notifications. I agree that we should probably make it for both money requests and expense reports. Thanks for clarifying! I can make a note of that in the other issue.

**Comment by mountiny** [comment]
Good shout, I think we concluded that Submitting an Open report which has a smartscanning expense should be fine, because with the changes to the edit rights, the submitter will be able to edit the smartscanned receipt that is in Processing state.

So that was about handling the expenses when submitting them which we do not have to worry about. 

I agree with @jasperhuangg then this issue should not be on hold for https://github.com/Expensify/App/issues/35240 sorry for not connecting the dots earlier

**Comment by youssef-lr** [comment]
@mountiny I think the issue here is that when:

1. User A scans a receipt with User B
2. User A sends a manual request to User B.
3. User B pays the report (which contains a receipt still being scanned)
4. The expense containing a smartscanning a receipt is also marked as paid, when it should be moved to another report.

Do you think we should handle this on its own issue?

**Comment by mountiny** [comment]
I see, I think we already do all we need in this flow but only for case of Expense reports https://github.com/Expensify/Auth/blob/85a0892f73f325cda027df427372ecc336f937d3/auth/command/PayMoneyRequest.cpp#L136-L165 nothing for IOU reports

Seems like this issue could handle that case

**Comment by youssef-lr** [comment]
Ok cool, I can work on this update

**Comment by garrettmknight** [comment]
Updating to reflect this is off hold - Youssef has been sick this week so no update, but he'll get to it when he's back up and running. Likely next week.

**Comment by youssef-lr** [comment]
I'm getting to this one next week.

**Comment by youssef-lr** [comment]
Still planning to get to this this week. I'm just getting back from OOO as I was sick.

**Comment by youssef-lr** [comment]
No update yet, will work on this this week.

**Comment by garrettmknight** [comment]
Update to note that we're focused on some higher priority work for the quarterly release so we'll get to this when that's done.

**Comment by youssef-lr** [comment]
No update yet.

**Comment by youssef-lr** [comment]
Will work on this soon.

**Comment by youssef-lr** [comment]
No update yet. Still busy with uneven splits.

**Comment by youssef-lr** [comment]
I'll be able to look into this next week.

**Comment by youssef-lr** [comment]
No update yet, still planning to get to this.

**Comment by youssef-lr** [comment]
I'll have time for this next week.

**Comment by youssef-lr** [comment]
No update yet, I'm focusing on fixing a few bugs for split affecting quality

**Comment by youssef-lr** [comment]
No update yet

**Comment by youssef-lr** [comment]
Same

**Comment by youssef-lr** [comment]
I think this should be fixed since we started moving scanning expenses to new reports when a report is paid. Going to try reproducing this week or the next one.

**Comment by youssef-lr** [comment]
This is fixed.
